.gitlab-ci.yml

```diff
@@ -149,29 +149,40 @@ variables:
 #
 # Condition mixins for simplification of rules
 #
-# FIXME: currently, the only good way we have to compose rules is by using YAML anchors.
-# However, as they are YAML features (and not Gitlab features), they have to be present
-# in each file that needs them, which results in a lot of copy-pasting.
-# In the future, we may be able to leverage YAML functions (https://gitlab.com/groups/gitlab-org/-/epics/4381)
-# and cross-file references (https://gitlab.com/gitlab-org/gitlab/-/issues/266173) to remove this
-# currently needed copy-paste.
-#
-# The list of all available rules is given below. To use a rule in a specific file, copy it.
-#
-
 .if_master_branch: &if_master_branch
   if: $CI_COMMIT_BRANCH == "master"
 
 .if_not_master_branch: &if_not_master_branch
   if: $CI_COMMIT_BRANCH != "master"
 
+.if_release_branch: &if_release_branch
+  if: $CI_COMMIT_BRANCH =~ /^[0-9]+\.[0-9]+\.x$/
+
+.if_version_6: &if_version_6
+  if: $RELEASE_VERSION_6 != ""
+
+.if_not_version_6: &if_not_version_6
+  if: $RELEASE_VERSION_6 == ""
+
+.if_version_7: &if_version_7
+  if: $RELEASE_VERSION_7 != ""
+
+.if_not_version_7: &if_not_version_7
+  if: $RELEASE_VERSION_7 == ""
+
+.if_deploy: &if_deploy
+  if: $DEPLOY_AGENT == "true"
+
 .if_tagged_commit: &if_tagged_commit
   if: $CI_COMMIT_TAG != null
 
-.if_not_tagged_commit: &if_not_tagged_commit
-  if: $CI_COMMIT_TAG == null
+.if_not_nightly_repo_branch: &if_not_nightly_repo_branch
+  if: $DEB_RPM_BUCKET_BRANCH != "nightly"
 
-# Rules to run all builds conditionally.
+.if_not_stable_or_beta_repo_branch: &if_not_stable_or_beta_repo_branch
+  if: $DEB_RPM_BUCKET_BRANCH != "beta" && $DEB_RPM_BUCKET_BRANCH != "stable"
```

Comment:
Isn't this the same as `$DEB_RPM_BUCKET_BRANCH == "nightly"` because `DEB_RPM_BUCKET_BRANCH` can take other values than `beta`, `stable` and `nightly`? `none` maybe? Or we never need `none` now that we have a separate `DEPLOY_AGENT`?

Could we simplify some conditions if we sanitized the inputs (eg: so we know we can't get `$DEB_RPM_BUCKET_BRANCH == "potato"`)?

Reply:
> Could we simplify some conditions if we sanitized the inputs (eg: so we know we can't get `$DEB_RPM_BUCKET_BRANCH == "potato"`)?

Yes. We can't actually prevent people from trying to run pipelines with arbitrary variables, so to me the only way  to sanitize is to add: `  && ($DEB_RPM_BUCKET_BRANCH == "nightly" || $DEB_RPM_BUCKET_BRANCH == "beta" || $DEB_RPM_BUCKET_BRANCH == "stable")` to https://github.com/DataDog/datadog-agent/blob/ed17d42ca3cb37ef3c68dc60f675caa60cbd9b0b/.gitlab-ci.yml#L46-L48

`none` was still supported (there's a job that relies on it: https://github.com/DataDog/datadog-agent/blob/ed17d42ca3cb37ef3c68dc60f675caa60cbd9b0b/.gitlab/check_deploy.yml#L36-L45), but I think we can remove that.

Reply:
But that won't change the number of conditions. The only change we can do is rename the `on_deploy_stable_or_beta_repo_branch` ruleset to `on_deploy_not_nightly_repo_branch`, but I think I like the former later, as it represents more precisely what the ruleset is about.

Reply:
Given that limiting valid pipeline branch won't have a direct effect on rules, I'll fo that on a follow-up PR.